cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md

```diff
@@ -10,13 +10,21 @@ It should be noted that defense mechanisms are intended to be used in a layered
 
 In many cases, brute force protections will overlap with credential stuffing defenses.
 
+Keep in mind that application can have different security levels for different users/roles or actions. For example for casual customers multi-factor authentication may by optional but for admins or some other special roles it can be required.
+
 ## Defense Option 1: Multi-Factor Authentication
 
 True multi-factor authentication is the best defense against Credential Stuffing attacks, but can have significant deployment and usability impacts, and so is frequently not a feasible option. If this defense is not feasible for your application, consider adopting as many of these other defenses as you can.
 
-## Defense Option 2: Multi-Step Login Process
+In order to balance security and usability multi-factor authentication can be combined with other techniques to ask for 2nd factor only in special situations. For example it can be combined with [device fingerprinting](cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md#defense-option-4-device-fingerprinting), in that scenario application can ask for 2nd factor only when application is accessed by unknown, new browser. Similary, cookies can be also used for remembering known browsers or application can ask about another factor based on login success ratio, IP address (like users from different network than company one like remote workers) etc.
 
-*Most of the automated account validation we've seen is using single step validation and checking for a success conditions. By forcing the client to render the response and include that in the next request (and including [Synchronizer (CSRF) Tokens](Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md)), we are just eliminating the basic attempts. It's not comprehensive.*
+Assuming having user e-mail address - asking user to retype short, random token from e-mail only after new sucessfull login on a new browser provides good balance bewteen security and usability in a lot of cases.
```

Comment:
Typo on `bewteen` .

In credentials stuffing does the attacker does not know the email (often used as login) of the targeted account? 

Most of the time email is considered as a "public" data (can be found via google dorking for example) so, it is not preferable to ask for random chunk of a private information that only the user know or possess or is ?

Example: 
* Specific characters on his second name,
* Specific characters of his social security number,
* ...

Reply:
It's not about the email in here, more than the token sent to their email and used to validate the browser. Credential stuffing is about the combination of the emails and the passwords.

Reply:
I think that is me that have not understood the sentence, so, correct me if i'm wrong:

When a user perform a successful authentication from a unknow browser then the system ask him to enter a random token that he has received in a parallel email sent, like an OTP, in order to force the attacker to have the control of the targeted user's mailbox in order to fully login into a system.

It is the expected meaning of the sentence?

Reply:
Yes, I should rephrase it probably.

Reply:
Do you have some ideas how to better express it?

Reply:
I think that just describe the approach you have in mind is enough.

Reply:
I decided that I will delete this sentence complete. This is partially repetition from previous paragraph where I tried to show the simplest 2fa schema. I think that it will be better to create new Multi-Factor Authentication CS where there will be description of forms of multi-factor and example how to implement it.
It will be possible to reference that CS here and in other CSs like Authentication Cheat Sheet or Access Control Cheat Sheet.